src/suspense/component.tsx

```diff
@@ -23,43 +60,46 @@ export class LazySuspense extends Component<
     setFallback: (fallback: Fallback) => {
       if (this.hydrationFallback === fallback) return;
       this.hydrationFallback = fallback;
-      this.useSibling = Boolean(fallback);
       // Schedule an update so we force switch from the sibling tree
       // back to the suspense boundary
-      this.forceUpdate();
+      if (this.mounted) this.forceUpdate();
     },
   };
 
   private hydrationFallback: Fallback = null;
-  private useSibling = false;
   private mounted = false;
 
   componentDidMount() {
     this.mounted = true;
   }
 
-  private DynamicFallback = ({
+  private DynamicFallback: FunctionComponent<DynamicFallbackProps> = ({
     children,
-    sibling,
-  }: {
-    children(fallback: Fallback): Fallback;
-    sibling: boolean;
+    outsideSuspense,
   }) => {
-    return (
-      <>
-        {sibling
-          ? children(this.useSibling ? this.hydrationFallback : null)
-          : children(this.useSibling ? null : this.props.fallback)}
-      </>
-    );
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    useLayoutEffect(() => {
+      return () => {
+        // the effect cleanup is called by the Suspense boundary itself
+        // when both Lazy AND the eventual promises thrown are done
+        // so Suspense will re-render with actual content and we remove
+        // the hydration fallback at the same time
+        if (!outsideSuspense) this.state.setFallback(null);
```

Comment:
Let Suspense decide when  we are ready to remove the SSR content and render the reactive one, so we can throw other promises (eg for data fetching) after bundle loaded and still maintain SSR content in the meantime (prev implementation switches to Suspense fallback instead)